### PR TITLE
Aw/disable flattened indices

### PR DESCRIPTION
**Pull request by averywpx**
Resolved the issue occurred in #280 by disabling the flattened names in `properties` as well.

### CODE REVIEW

File: hail_scripts/elasticsearch/elasticsearch_utils.py [hunk 1]
```diff
@@ -100,11 +100,17 @@ def elasticsearch_schema_for_table(table, disable_doc_values_for_fields=(), disa
         logger.info("==> will disable index fields for %s", ", ".join(disable_index_for_fields))
         for es_field_name in disable_index_for_fields:
             if es_field_name not in properties:
-                raise ValueError(
-                    "'%s' in disable_index_for_fields arg is not in the elasticsearch schema: %s"
-                    % (es_field_name, properties)
-                )
-            properties[es_field_name]["index"] = False
+                flattened_fields = [key for key in properties if f"{es_field_name}_" in key]
```
Comment (hanars) [inline hail_scripts/elasticsearch/elasticsearch_utils.py:103]: So the key has to start with the prefix not just contain it, so it should not be `f"{es_field_name}_" in key` it should be `key.startswith(f'{es_field_name}_')

**Review by hanars — CHANGES_REQUESTED** [review]

**Comment by hanars** [comment]
looks good t me, @mike-w-wilson thoughts?

**Comment by mike-w-wilson** [comment]
This looks good to me as well!

**Review by mike-w-wilson — APPROVED** [review]

**Review by hanars — APPROVED** [review]

**Comment by mike-w-wilson** [comment]
@averywpx , @hanars I've tested this on our airflow2 instance and have gotten this error for `rg37_locus`, `sortedTranscriptConsequences` and now `genotypes`. Should I just continue removing the disable_index=True? This seems like it may be an issue outside of the nested fields?
```
'Mapping definition for [genotypes] has unsupported parameters:  [index : false]')
```

**Comment by hanars** [comment]
`genotypes` is also a nested field so yeah I would just remove `disable_index=True` for that. I checked a successful index mapping, and only `sortedTranscriptConsequences` and  `genotypes` have type "nested". `rg37_locus` actually doesn't have a type at all, it has a nested "properties" dict that has the type for each sub-field, I guess because it is a single nested object instead of a list of nested objects (the 2 nested fields also have this "properties" dict). There are no other fields with these nested properties, so I think these should be the only affected fields

**Comment by mike-w-wilson** [comment]
So the index grew ~20% larger..... the 2/16/2022 index is 21.2gb, today's is 27.4gb. I confirmed there is the same number of records in both so there must be something else going on.

**Comment by mike-w-wilson** [comment]
Also, I just reran export, so this is the same variant and project MTs as the previous index

**Comment by mike-w-wilson** [comment]
And now I'm a liar -- the data management page is now showing 21 GB for today... 😕

**Comment by hanars** [comment]
so we're saying this doesn't actually save any space? Or that we aren't testing properly

**Comment by mike-w-wilson** [comment]
It looks like it saved a tiny bit of space, 0.2 GB for this index. I'm not sure why it showed a larger index at first. Maybe it was reporting the size during a temp to data node transfer or something.

**Comment by mike-w-wilson** [comment]
I've pushed a commit with the three fields disabled. Sorry for the confusion yesterday. The PR was tested properly. Even though it didn't save us a ton of space, looks like just 1%, I think we still merge this but would like to hear your thoughts @hanars on adding more complexity to the repo vs data savings. Also, I couldnt find if it is possible to disable indexing on nested field types in a different way but perhaps we would see larger savings if we could figure that out for genotypes in a different PR?

**Comment by hanars** [comment]
I would say my intuition is to merge this as is, because while there is added complexity I don't think it's too much to maintain and there is some data saving. I think for the 3 nested-ish fields, my intuition is we would need to set `"enabled": False` for each sub-field of the nested properties dict, which should only be a couple more lines of code in the handling of `disable_index_for_fields`. I hope those fields will add more noticeable space saving (they are very big) but if we would rather merge this working, tested code now and add that additional functionality as a separate PR that would make sense to me

**Pull request merged**